src/core_modules/capture-core/components/D2Form/field/configs/base/configBaseCustomForm.js

```diff
@@ -4,8 +4,9 @@ import { getBaseConfigForField } from './configBase';
 import type { DataElement as MetaDataElement } from '../../../../../metaData';
 import type { FieldConfigForType } from './configBase';
 
-const getBaseProps = (metaData: MetaDataElement) => ({
-    metaCompulsory: metaData.compulsory,
+const getBaseProps = ({ compulsory, disabled, unique }: MetaDataElement) => ({
+    metaCompulsory: compulsory,
+    metaDisabled: disabled || (unique && unique.generatable),
```

Comment:
NIT: You can use optional chaining: `metaDisabled: disabled || unique?.generatable,`